src/app/top-bar/top-bar.component.ts

```diff
@@ -0,0 +1,54 @@
+import { Component, OnInit } from '@angular/core';
+import { UserService } from '../shared/service/user.service';
+import { Router } from '@angular/router';
+
+@Component({
+  selector: 'app-top-bar',
+  templateUrl: './top-bar.component.html',
+  styleUrls: ['./top-bar.component.scss']
+})
+export class TopBarComponent implements OnInit {
+  menuOpen = false;
+  collapsed = true;
+  query: string;
+  username: string;
+
+  constructor(private userService: UserService,
+              private router: Router) {}
+
+
+  // Is this clean ? something seems wrong
+  // having duplicate code in app.component.ts and top.component.ts
+  // We need to load username here but redirection to login page
+  // should be done by the app.component ?
```

Comment:
You should probably only do the redirect in app.component. For the username here you could use an Input() to pass the user down from the app component since this is a child of it.

Reply:
Ok I will change this !
Any other comment on this PR ?

Reply:
Everything else looks fine. Just make sure all of the tests are still passing.